src/struveH.jl

```diff
@@ -0,0 +1,73 @@
+
+struveH(v::Real, x::Real) = _struveH(float(v), float(x))
+struveH(v, x) = _H_integral(v, x)
+
+_struveH(v, x::AbstractFloat) = _H_integral(v, x)
+_struveH(v::Float32, x::Float32) = Float32(_struveH(Float64(v), Float64(x)))
+_struveH(v::Float16, x::Float16) = Float16(_struveH(Float32(v), Float32(x)))
+
+function _struveH(v, x::T) where T <: Float64
+    if struveK_large_arg_cutoff(v, x)
+        # use http://dlmf.nist.gov/11.2.E6
+        return struveK_large_argument(v, x) + bessely(v, x)
+    elseif struveH_power_series_cutoff(v, x)
+        return struveH_power_series(v, x)
+    else
+        return _H_integral(v, x)
+    end
+end
+
+struveK(v::Real, x::Real) = _struveK(float(v), float(x))
+struveK(v, x) = _K_integral(v, x)
```

Comment:
I think this sort of dispatch could be done in `src/Struve.jl` by defining multiple methods for `Struve.K` etc. I guess this raises the question: Is `Struve.K` okay for users or do we need to export something like `struvek` (in analogy to `besselj` from SpecialFunctions)

Reply:
I'm generally supported of exported `struvek` and `struveh` in the `src/Struve.jl` just to keep the ecosystem consistent. There has been some discussion on this at SpecialFunctions.jl for other functions and I think the general consensus is lower case for all the function similar to `besselk`. 

What do you think? I find the naming in the files when looking through the source easier to reason about if the function is `struvek` instead of `k` but that is probably because I'm also looking at other sources at the same time with `besselk` :smile:

Reply:
I agree, let's export `struvek`, `struvem` etc so that we're consistent with SpecialFunctions.jl . Accordingly, we should remove the constants for `Struve.K`, `Struve.M` etc (and bump the version number)

Reply:
Sounds great! I've changed these function definitions, added significant more docs and links to NIST, and incremented version. I will sort through the constant definitions and make sure I didn't miss anything this afternoon!

Reply:
Looks like I need to look through the subnormal cases as they sometimes fail. I’ll have time in the afternoon to sort those out.